// =============================================================================
// Copyright (C) 2022 Xiaomi EV Company Limited. All rights reserved.
//
// XIAOMI CONFIDENTIAL
//
// Xiaomi EV Company Limited retains all intellectual property and proprietary
// rights in and to this software and related documentation and any
// modifications thereto ("Software"). Any reproduction, disclosure or
// distribution of this Software is strictly prohibited.
//
// Permitted Users may only access, use or edit the Software based on their work
// duties. Permitted Users shall refer to the current employees in Xiaomi EV
// Autonomous Driving Group who have entered into Non-disclosure Agreement with
// Xiaomi, and have been duly authorized to access the Software.
// =============================================================================

syntax = "proto2";

package proto;

message CodeBlockExecution {
  optional string name = 1;
  optional bool is_code_block_conditional = 2;
  optional bool code_block_condition_satisfied = 3;
  optional float duration_ms = 4 [default = -1.0];
  repeated string details = 5;
  repeated CodeBlockExecution child_executions = 6;
}

message ObstacleCodePathReport {
  optional string name = 1;
  map<string, CodeBlockExecution> obstacle_wise_root_executions = 2;
}
